restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/ConferencesEndpoint.java

```diff
@@ -194,4 +207,77 @@ protected Response getConferences(final String accountSid, UriInfo info, MediaTy
+        try {
+            secure(account, "RestComm:Modify:Conferences");
+        } catch (final AuthorizationException exception) {
+            return status(UNAUTHORIZED).build();
+        }
+        final ConferenceDetailRecordsDao dao = daoManager.getConferenceDetailRecordsDao();
+        ConferenceDetailRecord cdr = dao.getConferenceDetailRecord(new Sid(sid));
+        if (cdr == null) {
+            return status(NOT_FOUND).build();
+        } else {
+            try {
+                secure(account, cdr.getAccountSid(), SecuredType.SECURED_STANDARD);
+            } catch (final AuthorizationException exception) {
+                return status(UNAUTHORIZED).build();
+            }
+
+            final String status = data.getFirst("Status");
+
+            if(status != null){
+                if(status.equalsIgnoreCase("completed")){
+                    kickoutAllActiveParticipants(cdr);
+                    //get updated conference record
+                    cdr = dao.getConferenceDetailRecord(new Sid(sid));
+                }
+            }
+            if (APPLICATION_XML_TYPE == responseType) {
+                final RestCommResponse response = new RestCommResponse(cdr);
+                return ok(xstream.toXML(response), APPLICATION_XML).build();
+            } else if (APPLICATION_JSON_TYPE == responseType) {
+                return ok(gson.toJson(cdr), APPLICATION_JSON).build();
+            } else {
+                return null;
+            }
+        }
+    }
+
+    private void kickoutAllActiveParticipants(ConferenceDetailRecord conferenceDetailRecord){
+        List<CallDetailRecord> callDetailRecords = daoManager.getCallDetailRecordsDao().getRunningCallDetailRecordsByConferenceSid(conferenceDetailRecord.getSid());
+
+        if(callDetailRecords == null || callDetailRecords.isEmpty()){
+            if (logger.isInfoEnabled())
+                logger.info("no active participants found.");
+        } else {
+           try {
+               if (logger.isInfoEnabled())
+                    logger.info("total conference participants are: "+callDetailRecords.size());
+               Iterator<CallDetailRecord> iterator = callDetailRecords.iterator();
+                while(iterator.hasNext()){
+                    ActorRef callApiClient = callApiClient(iterator.next().getSid());
+                    callApiClient.tell(new Hangup(), null);
+                }
+            } catch (Exception e) {
+                logger.error("Exception while trying to terminate conference via api: ", e);
+            }
+        }
+    }
+
+    private ActorRef callApiClient(final Sid sid){
+        final Props props = new Props(new UntypedActorFactory() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public UntypedActor create() throws Exception {
+                return new CallApiClient(sid, daoManager);
+            }
+        });
+        return system.actorOf(props);
```

Comment:
@maria-farooq here you create a top-level actor with a parent the system itself.

Nowhere in the patch, you destroy this actor, thus this actor and its child actors `HttpAsycClientHelper` will stay in memory for as long as the akka system is up and running.

Reply:
@maria-farooq I believe better design would be to use ASK pattern, delegate the task to CallManager which create CallApiClient , use to kill the calls and then destroy the actor.

On top, we can use ASK pattern without blocking, just fire the request and return, since we don't need the response.

Reply:
@maria-farooq also delegating the task to CallManager, we can save some LCM (http) requests.

CallManager can check which of the CDRs are local calls or not and either use update call method or LCM.

Reply:
@maria-farooq I see the local calls are removed by the Conferences endpoint (Evicting private class) but still I think the task to remove remote calls actors should be delegated to CallManager that will create CallApiClient actors and then destroy them